contrib/Dockerfile

```diff
@@ -1,10 +1,48 @@
-# This file is used to add the nightly Dgraph binaries and assets to Dgraph base
-# image.
+# This file builds the dgraph binary and dgraph docker image.
 
-# This gets built as part of release.sh. Must be run from /tmp/build, with the linux binaries
-# already built and placed there.
+# This gets built as part of our CD pipeline. Must be run through Makefile.
 
-FROM ubuntu:20.04
+FROM --platform=linux/amd64 ubuntu:20.04 AS build
+
+RUN mkdir /dgraph
+WORKDIR /dgraph
+
+COPY . ./
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
+    curl \
+    ca-certificates
+
+RUN GO_VERSION=$({ [ -f .go-version ] && cat .go-version; }) \
+    && curl -O -L "https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz" \
+    && tar -C /usr/local/ -xzf go${GO_VERSION}.linux-amd64.tar.gz \
+    && rm -f go${GO_VERSION}.linux-amd64.tar.gz
+
+ENV PATH=$PATH:/usr/local/go/bin:/root/go/bin \
+    GOPATH=/root/go
+
+# save time by caching this layer
+RUN cd dgraph && make jemalloc
+
+ARG BUILD
```

Comment:
where are these args used?

Reply:
we cannot execute git commands inside of the container (since it is a very lightweight image is does not include `git`).  However part of our build process includes invoking `git` to fetch info about what branch we are on, commit SHA, etc (this is in `Makefile` and `dgraph/Makefile`).  So it is not possible to do this inside the container.  This workaround means that these parameters are passed inside the container as docker arguments.

Reply:
we can install git, that is a throw away container anyway instead of figuring out new workarounds.